common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -564,20 +564,25 @@ public TokenResult renewAccessToken(@NonNull final SilentTokenCommandParameters
                     "Token request was successful"
             );
 
-            // Remove old Access Token
-            Logger.info(
-                    methodTag,
-                    "Access token is refresh-expired. Removing from cache..."
-            );
-            final AccessTokenRecord accessTokenRecord = cacheRecord.getAccessToken();
-            cache.removeCredential(accessTokenRecord);
+            List<ICacheRecord> savedRecords;
```

Comment:
I'm not sure if this code path is even exercised at all.

This one seems to be invoked by the code path that relies on LibraryConfiguration.refreshInEnabled, which seems to be set to OFF by default.

Probably worth pursuing if this one is in use. Shane or Olga might be able to give you the answer. @shahzaibj do you remember? My memory around this feature is very very faint.

If not (and it's never going to be used) - maybe removing the code might be a better way to go.

Reply:
I remember reviewing those PRs long time back along with Shane. The memory is faint but the parts I remember is that the feature was sort of opt-in and like you said it is OFF by default. Possible that there are customers who have enabled it and are getting benefits already. 

This was the PR: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1310

The feature is focused on resiliency so I think we certainly need it. Interestingly I had someone reach out today asking for something similar to this feature so I definitely think there is partner and customer desire for it as well. Let's keep the code as we likely need now and in the future to continue to build resiliency in our products.